core/vm/contracts.go

```diff
@@ -79,6 +83,10 @@ var PrecompiledContractsIstanbul = map[common.Address]PrecompiledContract{
 
 	common.BytesToAddress([]byte{100}): &tmHeaderValidate{},
 	common.BytesToAddress([]byte{101}): &iavlMerkleProofValidate{},
+
+	//Precompiled contracts for sha3-256 FIPS202 & recover uncompressed public key
+	common.BytesToAddress([]byte{102}): &sha3fips{},
```

Comment:
We should have another variable for this hardfork, like `PrecompiledContracts Hilbert `

Reply:
Updated this to have another hard fork as Hilbert.
along with this fix, rebased to latest and with some fixes, moved the testing data to the json file test data